rclpy/rclpy/node.py

```diff
@@ -12,40 +12,131 @@
+        if srv_type.__class__._TYPE_SUPPORT is None:
+            srv_type.__class__.__import_type_support__()
+        if srv_type.__class__._TYPE_SUPPORT is None:
+            raise NoTypeSupportImportedException
+        [client_handle, client_pointer] = _rclpy.rclpy_create_client(
+            self.handle,
+            srv_type,
+            srv_name,
+            qos_profile.get_c_qos_profile())
+        client = Client(
+            self.handle, client_handle, client_pointer, srv_type, srv_name, qos_profile)
+        self.clients.append(client)
+        return client
+
+    def create_service(
+            self, srv_type, srv_name, callback, qos_profile=qos_profile_services_default):
+        if srv_type.__class__._TYPE_SUPPORT is None:
+            srv_type.__class__.__import_type_support__()
+        if srv_type.__class__._TYPE_SUPPORT is None:
+            raise NoTypeSupportImportedException
+        [service_handle, service_pointer] = _rclpy.rclpy_create_service(
+            self.handle,
+            srv_type,
+            srv_name,
+            qos_profile.get_c_qos_profile())
+        service = Service(
+            self.handle, service_handle, service_pointer,
+            srv_type, srv_name, callback, qos_profile)
+        self.services.append(service)
+        return service
+
+    def destroy_publisher(self, publisher):
+        for pub in self.publishers:
+            if pub.publisher_handle == publisher.publisher_handle:
+                _rclpy.rclpy_destroy_node_entity(
+                    'publisher', pub.publisher_handle, self.handle)
+                self.publishers.remove(pub)
+                return True
+        return False
+
+    def destroy_subscription(self, subscription):
+        for sub in self.subscriptions:
+            if sub.subscription_handle == subscription.subscription_handle:
+                _rclpy.rclpy_destroy_node_entity(
+                    'subscription', sub.subscription_handle, self.handle)
+                self.subscriptions.remove(sub)
+                return True
+        return False
+
+    def destroy_service(self, service):
+        for srv in self.services:
+            if srv.service_handle == service.service_handle:
+                _rclpy.rclpy_destroy_node_entity(
+                    'service', srv.service_handle, self.handle)
+                self.services.remove(srv)
+                return True
+        return False
+
+    def destroy_client(self, client):
+        for cli in self.clients:
+            if cli.client_handle == client.client_handle:
+                _rclpy.rclpy_destroy_node_entity(
+                    'client', cli.client_handle, self.handle)
+                self.clients.remove(cli)
+                return True
+        return False
+
     def get_topic_names_and_types(self):
-        return rclpy._rclpy.rclpy_get_topic_names_and_types(self.handle)
+        return _rclpy.rclpy_get_topic_names_and_types(self.handle)
+
+    def __del__(self):
```

Comment:
Sounds good to me: https://github.com/ros2/rclpy/pull/51/commits/822a71c0ba3f99c08dd28ba58319b539d559d646